### PR TITLE
feat: Vercel source plugin

### PR DESCRIPTION
**Pull request by disq**
- [x] Read / auto-detect teams in Configure
- [x] Domains
- [x] Domain Records
- [x] Teams
- [x] Team Members
- [x] Deployments
- [x] Deployment Checks
- [x] Projects
- [x] Project Environments

### CODE REVIEW

File: plugins/source/vercel/.pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+repos:
```
Comment (erezrokah) [inline plugins/source/vercel/.pre-commit-config.yaml:1]: Let's remove this and the relevant make target, see https://github.com/cloudquery/cloudquery/pull/5559

File: plugins/source/vercel/Makefile [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+.PHONY: install-tools
+install-tools:
+	@echo Installing tools from tools/tool.go
+	@cat tools/tool.go | grep _ | awk -F'"' '{print $$2}' | xargs -tI % go install %
+
+# Install pre-commit hooks. This requires pre-commit to be installed (https://pre-commit.com/)
+.PHONY: install-hooks
+install-hooks:
+	pre-commit install
+
```
Comment (erezrokah) [inline plugins/source/vercel/Makefile:21]:
```suggestion
```

See https://github.com/cloudquery/cloudquery/pull/5559

File: plugins/source/vercel/README.md [hunk 3]
```diff
@@ -0,0 +1,7 @@
+# Vercel Plugin
```
Comment (erezrokah) [inline plugins/source/vercel/README.md:1]: We'll need to add `Vercel` to https://github.com/cloudquery/cloudquery/blob/51c68f8e934192a27ac6250a66df62e2b06d9797/.github/styles/Vocab/Base/accept.txt

**Review by erezrokah — APPROVED** [review]
🚀

File: plugins/source/vercel/internal/vercel/vercel.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+		if res.StatusCode == http.StatusTooManyRequests {
+			val := res.Header.Get("X-Ratelimit-Reset")
+			if val != "" {
+				t, err := strconv.ParseInt(val, 10, 64)
+				if err == nil && t > 0 {
+					ts := time.Unix(t, 0)
+					val = ts.Format(time.RFC3339) + fmt.Sprintf(" (in %s)", time.Until(ts).Round(time.Second))
+				}
+
+				return nil, fmt.Errorf("request to %s failed: %s. Rate limit will reset at: %s", path, res.Status, val)
```
Comment (erezrokah) [inline plugins/source/vercel/internal/vercel/vercel.go:83]: Probably a follow up, but we can add retry logic if the reset window is not large

File: plugins/source/vercel/README.md (hunk 3 above)
Comment (disq) [reply plugins/source/vercel/README.md:1]: Yeah, it will end with merge conflicts and pain so I'm waiting until the review is over

**Review by yevgenypats — CHANGES_REQUESTED** [review]
Looks good. One major comment and then I'll give it another round of reviews:

Can we avoid using mocks here and use the same approach we now use in GCP and Azure with a test http server and providing a test `baseURL` or a test `httpClient` ? This should reduce the amount of boilerplate, make it inline with all other plugins and give even better test coverage.

File: plugins/source/vercel/Makefile [hunk 5]
```diff
@@ -0,0 +1,29 @@
+# Test unit
+.PHONY: test
+test:
+	go test -timeout 3m ./...
+
+# Install tools
+.PHONY: install-tools
+install-tools:
+	@echo Installing tools from tools/tool.go
```
Comment (yevgenypats) [inline plugins/source/vercel/Makefile:9]: I think this is deprecated given mockgen is not used.

**Review by yevgenypats — APPROVED** [review]
Nice! Looks great. Had one nit.

**Pull request merged**